### PR TITLE
Added CourseState change endpoint.

### PR DESCRIPTION
**Pull request by waheedahmed**
[ECOM-7004](https://openedx.atlassian.net/browse/ECOM-7004)

Created endpoint to change course workflow state.

### CODE REVIEW

**Comment by waheedahmed** [comment]
@awais786, @tasawernawaz, @bderusha Please review.

File: course_discovery/apps/publisher/api/tests/test_serializers.py [hunk 1]
```diff
@@ -4,9 +4,12 @@
 
 from course_discovery.apps.core.tests.factories import UserFactory
 from course_discovery.apps.publisher.api.serializers import (
-    CourseUserRoleSerializer, GroupUserSerializer, UpdateCourseKeySerializer
+    CourseUserRoleSerializer, GroupUserSerializer, UpdateCourseKeySerializer,
+    CourseStateSerializer
```
Comment (bderusha) [inline course_discovery/apps/publisher/api/tests/test_serializers.py]: NIT: this fits onto the line above

File: course_discovery/apps/publisher/api/urls.py [hunk 2]
```diff
@@ -2,12 +2,18 @@
...
+    CourseRoleAssignmentView, OrganizationGroupUserView, UpdateCourseKeyView,
+    ChangeCourseStateView
 )
 
 urlpatterns = [
     url(r'^course_role_assignments/(?P<pk>\d+)/$', CourseRoleAssignmentView.as_view(), name='course_role_assignments'),
     url(r'^admins/organizations/(?P<pk>\d+)/users/$', OrganizationGroupUserView.as_view(),
         name='organization_group_users'),
+    url(
+        r'^courses/(?P<course_id>\d+)/change_state/(?P<pk>\d+)/$',
```
Comment (bderusha) [inline course_discovery/apps/publisher/api/urls.py]: What is the `pk` here?

File: course_discovery/apps/publisher/api/serializers.py [hunk 3]
```diff
@@ -71,3 +72,28 @@ def update(self, instance, validated_data):
...
         return instance
+
+
+class CourseStateSerializer(serializers.ModelSerializer):
+    """Serializer for the `CourseState` model to change role assignment. """
+
+    class Meta:
+        model = CourseState
+        fields = ('name', 'approved_by_role', 'owner_role', 'course',)
+        read_only_fields = ('course', 'approved_by_role', 'owner_role')
```
Comment (bderusha) [inline course_discovery/apps/publisher/api/serializers.py]: This is deprecated in DRF 3.x

Please use the `extra_kwargs` field instead http://www.django-rest-framework.org/api-guide/serializers/#additional-keyword-arguments

File: course_discovery/apps/publisher/api/urls.py (hunk 2 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/api/urls.py]: Primary key of `CourseState` models object, which we are actually updating.

**Comment by waheedahmed** [comment]
@bderusha Feedback addressed, kindly take another look.

File: course_discovery/apps/publisher/api/tests/test_serializers.py [hunk 4]
```diff
@@ -71,3 +73,27 @@ def test_validation_error(self):
...
+        self.assertNotEqual(self.course_state, CourseStateChoices.Review)
+        serializer = self.serializer_class(self.course_state)
+        data = {'name': CourseStateChoices.Review}
+        serializer.update(self.course_state, data)
+
+        self.course_state = CourseState.objects.get(course=self.course_state.course)
+        self.assertEqual(self.course_state.name, CourseStateChoices.Review)
+
+    def test_update_with_error(self):
+        serializer = self.serializer_class(self.course_state)
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/api/tests/test_serializers.py]: add doctstring

File: course_discovery/apps/publisher/api/tests/test_serializers.py [hunk 5]
```diff
@@ -71,3 +73,27 @@ def test_validation_error(self):
...
+
+
+class CourseStateSerializerTests(TestCase):
+    serializer_class = CourseStateSerializer
+
+    def setUp(self):
+        super(CourseStateSerializerTests, self).setUp()
+        self.course_state = CourseStateFactory(name=CourseStateChoices.Draft)
+
+    def test_update(self):
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/api/tests/test_serializers.py:139]: add doctstring

File: course_discovery/apps/publisher/tests/test_model.py [hunk 6]
```diff
@@ -425,6 +425,13 @@ def test_approved(self):
 
         self.assertEqual(self.course_state.name, CourseStateChoices.Approved)
 
+    def test_change_state(self):
+        self.assertNotEqual(self.course_state.name, CourseStateChoices.Review)
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/tests/test_model.py]: add docstring

**Review by tasawernawaz — APPROVED** [review]
Need docstring in some tests.
Do we need to add test to test the change state from `review` to `draft `?

File: course_discovery/apps/publisher/api/tests/test_serializers.py (hunk 5 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/api/tests/test_serializers.py:139]: Added.

**Review by tasawernawaz — APPROVED** [review]

**Review by bderusha — APPROVED** [review]

File: course_discovery/apps/publisher/api/urls.py (hunk 2 above)
Comment (bderusha) [reply course_discovery/apps/publisher/api/urls.py]: If we know the ID of the change_state object do we need the `course_id` at all?

Comment (waheedahmed) [reply course_discovery/apps/publisher/api/urls.py]: @bderusha Done! removed `course_id` from endpoint.

**Pull request merged**